lib/rucio/tests/test_heartbeat.py

```diff
@@ -70,5 +71,29 @@ def test_heartbeat_2(self):
         die('test0', 'host2', pids[2], threads[2])
         assert_equal(live('test0', 'host3', pids[3], threads[3]), {'assign_thread': 1, 'nr_threads': 2})
 
+    def test_heartbeat_payload(self):
+        """ HEARTBEAT (CORE): Test heartbeat with payload"""
+
+        pids = [self.__pid() for _ in range(6)]
+        threads = [self.__thread() for _ in range(6)]
+
+        live('test5', 'host0', pids[0], threads[0], payload='payload1')
```

Comment:
what about 

`payload={'key1':'value1','key2':'value2'}`
vs
`payload={'key2':'value2','key1':'value1'}`

Reply:
That wouldn't work, it's just a string. You could serialise the dictionary though, but it won't be handled specially in the heartbeat module.